scripts/Adapters/DeployArbAdapter.sol

```diff
@@ -1,42 +1,94 @@
 // SPDX-License-Identifier: BUSL-1.1
 pragma solidity ^0.8.0;
 
+import './BaseAdapterScript.sol';
 import {ArbAdapter, IArbAdapter} from '../../src/contracts/adapters/arbitrum/ArbAdapter.sol';
 import {ArbitrumAdapterTestnet} from '../contract_extensions/ArbitrumAdapter.sol';
-import './BaseAdapterStructs.sol';
 
 library ArbAdapterDeploymentHelper {
   struct ArbAdapterArgs {
-    BaseAdapterStructs.BaseAdapterArgs baseArgs;
+    BaseAdapterArgs baseArgs;
     address inbox;
     address destinationCCC;
   }
 
   function getAdapterCode(ArbAdapterArgs memory arbArgs) internal pure returns (bytes memory) {
-    if (arbArgs.baseArgs.isTestnet) {
-      return
-        abi.encodePacked(
-          type(ArbitrumAdapterTestnet).creationCode,
-          abi.encode(
-            arbArgs.baseArgs.crossChainController,
-            arbArgs.inbox,
-            arbArgs.destinationCCC,
-            arbArgs.baseArgs.providerGasLimit,
-            arbArgs.baseArgs.trustedRemotes
-          )
+    bytes memory creationCode = arbArgs.baseArgs.isTestnet
+      ? type(ArbitrumAdapterTestnet).creationCode
+      : type(ArbAdapter).creationCode;
+
+    return
+      abi.encodePacked(
+        creationCode,
+        abi.encode(
+          arbArgs.baseArgs.crossChainController,
+          arbArgs.inbox,
+          arbArgs.destinationCCC,
+          arbArgs.baseArgs.providerGasLimit,
+          arbArgs.baseArgs.trustedRemotes
+        )
+      );
+  }
+}
+
+abstract contract BaseDeployArbAdapter is BaseAdapterScript {
+  function INBOX() internal view virtual returns (address) {
+    return address(0);
+  }
+
+  function _getConstructorArgs(
+    address crossChainController,
+    IBaseAdapter.TrustedRemotesConfig[] memory trustedRemotes
+  ) internal view returns (ArbAdapterDeploymentHelper.ArbAdapterArgs memory) {
+    require(crossChainController != address(0), 'CCC needs to be deployed');
+
+    Addresses memory remoteAddresses;
+    if (isTestnet()) {
+      if (TRANSACTION_NETWORK() == TestNetChainIds.ETHEREUM_SEPOLIA) {
+        remoteAddresses = _getAddresses(TestNetChainIds.ARBITRUM_SEPOLIA);
```

Comment:
Maybe `_getAddresses` should not be a part of this repo, taking into account that `crossChainController` address we already provide as the function input